src/auth/UsersComponent.tsx

```diff
@@ -1,101 +1,187 @@
-import React, { useEffect, useState, useContext } from 'react';
-import config from '../config';
-import { Box, MenuItem, FormControl, Select } from '@mui/material';
-import { AuthContext } from './AuthContext';
+import React, { useEffect, useState, useContext } from "react";
+import config from "../config";
+import { Box, MenuItem, FormControl, Select } from "@mui/material";
+import { AuthContext } from "./AuthContext";
+import { useLoading } from "../spinner/LoadingProvider";
+import { Snackbar, Alert } from "@mui/material";
 
-type UserRole = 'ADMIN' | 'PLAYER';
+type UserRole = "ADMIN" | "PLAYER";
 
 type User = {
-    userId: string;
-    username: string;
-    role: UserRole;
+  userId: string;
+  username: string;
+  role: UserRole;
 };
 
 const UsersComponent: React.FC = () => {
-    const [users, setUsers] = useState<User[]>([]);
-    const [loading, setLoading] = useState(true);
-    const [error, setError] = useState<string | null>(null);
+  const [users, setUsers] = useState<User[]>([]);
+  const [error, setError] = useState<string | null>(null);
+  const [snackbar, setSnackbar] = useState({
+    open: false,
+    message: "",
+    severity: "success",
+  });
 
-    const { isAdmin } = useContext(AuthContext);
-    
-    useEffect(() => {
-        fetch(`${config.authServiceUrl}/public/users/users`)
-            .then((res) => {
-                if (!res.ok) throw new Error('Failed to fetch users');
-                return res.json();
-            })
-            .then((data) => {
-                setUsers(data);
-                setLoading(false);
-            })
-            .catch((err) => {
-                setError(err.message);
-                setLoading(false);
-            });
-    }, []);
+  const { isAdmin } = useContext(AuthContext);
+  const { setLoading: setGlobalLoading } = useLoading();
 
-    const handleRoleChange = (userId: string, newRole: UserRole) => {
-        console.log(`Changing role for user ${userId} to: ${newRole}`);
-        // Call API for a change role
-        setUsers(prevUsers => 
-            prevUsers.map(user => 
-                user.userId === userId ? { ...user, role: newRole } : user
-            )
-        );
-    };
+  useEffect(() => {
+    fetch(`${config.authServiceUrl}/public/users/users`)
```

Comment:
Shouldn't we have setLoading(true) just before fetch? Previously it was useState(true), now it's nowhere setting loading to true. Is spinner visible at all?